src/pdf/base/XojCairoPdfExport.cpp

```diff
@@ -153,20 +155,18 @@ auto XojCairoPdfExport::createPdf(fs::path const& file, PageRangeVector& range,
         return false;
     }
 
-    int count = 0;
-    for (PageRangeEntry* e: range) {
-        count += e->getLast() - e->getFirst() + 1;
-    }
+    size_t count = 0;
+    for (auto& e: range) { count += e.last - e.first + 1; }
 
     if (this->progressListener) {
         this->progressListener->setMaximumState(count);
     }
 
-    int c = 0;
-    for (PageRangeEntry* e: range) {
-        for (int i = e->getFirst(); i <= e->getLast(); i++) {
-            if (i < 0 || i >= static_cast<int>(doc->getPageCount())) {
-                continue;
+    size_t c = 0;
+    for (auto& e: range) {
+        for (size_t i = e.first; i <= e.last; i++) {
+            if (i >= doc->getPageCount()) {
+                break;
```

Comment:
Change of behavior at line 169. This makes this PR more than a refactor.

Reply:
This is actually quite the improvement to the algorithm. `continue` would test all pages of the document against this condition even though it is clear that any following page will not be part of the selection if the previous wasn't. `break` instead aborts the inner loop after the first violation of the condition.

As our documents do generally not have negative page numbers, there is no negative difference in the behavior compared to the previous version.